modules/uart_io.h

```diff
@@ -159,10 +177,45 @@ void uart_puts(char *s);
     * \retval true  Interrupt flag corresponding to the UART RX DMA channel.
     * \retval false RX DMA interrupt flag is not set.
     **/
-    bool is_uart_rx_dma_isr(void);
+    #include <msp430_xc.h>

```

Comment:
linker cant inline anithing, only compiler can. 
therefore this headers heave to be exposed at compile time, to allow compiler inline and optimise code deeply.